packages/yourloops/pages/main-layout.tsx

```diff
@@ -42,9 +52,46 @@ export function MainLayout(): JSX.Element {
     <React.Fragment>
       {session &&
         <NotificationContextProvider>
-          {session.user.role === UserRoles.caregiver && <CaregiverPage />}
-          {session.user.role === UserRoles.hcp && <HcpPage />}
-          {session.user.role === UserRoles.patient && <PatientPage />}
+          <TeamContextProvider>
+            <DataContextProvider context={DefaultDataContext}>
+              <MainHeader />
+              <Switch>
+                <Route exact path="/preferences" component={ProfilePage} />
+                <Route exact path="/notifications" component={NotificationsPage} />
+
+                {session.user.role === UserRoles.hcp &&
+                  <Switch>
+                    <Route exact path="/teams" component={TeamsPage} />
+                    <Route exact path="/certify" component={CertifyAccountPage} />
+                    <Route path="/patient/:patientId" component={PatientDataPage} />
```

Comment:
```
<Route path="/patient/:patientId" component={PatientDataPage} />
<Route exact path="/patients" component={PatientListPage} />
<Redirect exact from="/" to="/patients" />
<Redirect to="/not-found" />
```
This routes seem to be the same for a hcp and a caregiver. 
You could have a condition checking wether or not we have the hcp role or the caregiver role, and if so, define this routes. This would avoid duplicate code :slightly_smiling_face:

Reply:
Could not refactor at the moment because we still have the <CaregiverPage> component that fetch some data from a store. We will have to change this later with the new layout.